zds/forum/models.py

```diff
@@ -280,11 +296,55 @@ def last_read_post(self):
         except TopicRead.DoesNotExist:
             return self.first_post()
 
+    def resolve_last_read_post_absolute_url(self):
+        """resolve the url that leads to the last post the current user has read. If current user is \
+        anonymous, just lead to the thread start.
+
+        :return: the url
+        :rtype: str
+        """
+        user = get_current_user()
+        if not user.is_authenticated():
+            return self.resolve_first_post_url()
+        else:
+            return '{0}?page=1#p{1}'.format(
+                self.get_absolute_url(),
+                self.resolve_last_post_pk_read_by_user(user))
+
+    def resolve_last_post_pk_read_by_user(self, user):
+        """get the primary key of the last post the user read
+
+        :param user: the current (authenticated) user. Please do not try with unauthenticated user, il would lead to a \
+        useless request.
```

Comment:
(ce pourquoi je propose l'ajout d'un `if user and user.is_authenticated()`, on est jamais sur de rien)